### DISCUSSION TITLE
After flashing core-minimal-image the Jetson agx orin devkit fails to bootup and /dev/ttyACM0 disappears.

### DISCUSSION THREAD

**Discussion by rensejacob** [OE4T/meta-tegra#1230] [Q&A]
Hi, I had used the steps at https://koansoftware.com/yocto-project-on-nvidia-jetson-agx-xavier/ as reference to build and flash the core-minimal-image for jetson-agx-orin-devkit from Kirkstone branch. After flashing using the doflash.sh for jetson-agx-orin-devkit, the board refuses to boot up. I am not able to see the /dev/ttyACM0 interface after flashing the new image.
Even if I replace the Image build from OE4T in the boot/Image,  /dev/ttyACM0 fails to list. So I am not able to see the logs what is going wrong. Is there an alternative to see the serial logs if the kernel fails?

**Comment by ichergui** [comment]
Hello @rensejacob 
FYI, `Jetson Orin AGX` is not supported in `dunfell`
Please check our wiki page regarding [which branch you should use](https://github.com/OE4T/meta-tegra/wiki/Which-branch-should-I-use%3F) 
I recommend our [tegra-demo-distro](https://github.com/OE4T/tegra-demo-distro) instead and follow the steps mentioned [here](https://github.com/OE4T/tegra-demo-distro#tegra-demo-distro)

**Comment by ichergui** [comment]
Hello @rensejacob 
Any updates ?

**Reply by rensejacob** [reply]
@ichergui It seems to solve the problem I faced. Also I noted that after flashing the OE4T image when I do a lsusb it doesn't show NVIDIA L4T USB entry. 
I am also trying to upgrade the Kernel to 5.15. What are the changes required in recipes? I am not a Yocto expert. Appreciate the support.

**Reply by madisongh** [reply]
>  Also I noted that after flashing the OE4T image when I do a lsusb it doesn't show NVIDIA L4T USB entry.

We don't include that by default.  Just add the `l4t-usb-device-mode` package to your image if you want it.  For example, you could set `CORE_IMAGE_EXTRA_INSTALL = "l4t-usb-device-mode"` in your local.conf file.

> I am also trying to upgrade the Kernel to 5.15. What are the changes required in recipes?

The L4T and JetPack content are tightly coupled to the specific downstream kernel provided by NVIDIA.  Changing to a different kernel (or attempting to upgrade the NVIDIA kernel to a different base version) would likely lead to breakage of most of the important/interesting functionality in L4T, and would be a lot of work.  You'll find some pages on using upstream/mainline kernels on the [elinux.org](https://elinux.org) site, if you really want to try it, and you can consult the [documentation on the Yocto Project web site](https://docs.yoctoproject.org/) for more information on creating/updating recipes.
